modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceOrderManager.java

```diff
@@ -117,7 +119,8 @@ public void completeOrderProgress(UUID orderId, TaskStatus taskStatus, String er
         }
         serviceOrder.setTaskStatus(taskStatus);
         serviceOrder.setCompletedTime(OffsetDateTime.now());
-        serviceOrder.setErrorMsg(errorMsg);
+        serviceOrder.setError(ErrorResponse
```

Comment:
For all deployment failures, for example in https://github.com/eclipse-xpanse/xpanse/blob/f332609e15cac28b6181c1962603357b3306abaa/modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/migration/steps/ProcessDeploymentResult.java#L88

we can define a new ErrorType -> DeploymentFailedException